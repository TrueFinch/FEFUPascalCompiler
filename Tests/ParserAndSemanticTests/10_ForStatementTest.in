program a;
var
    i: integer;
begin
    for i := 0 downto 5 do begin
        write(i);
    end;
end.