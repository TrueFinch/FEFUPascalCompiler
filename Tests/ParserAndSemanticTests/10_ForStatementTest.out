                      a                       
                      ┼                       
                  MainBlock                   
       ┌──────────────┴──────┐                
      var            CompoundStatement        
       ┼                     ┼                
 SimpleVarDecl          ForStatement          
 ┌─────┼        ┌────┬───────┴─────┐          
 i  integer     i  downto  CompoundStatement  
                   ┌─┴┐      ┌─────┴          
                   0  5    write              
                           ┌─┴                
                           i                  
